### PR TITLE
Linter: add rule to remove default multipleOf: 1

### PR DESCRIPTION
**Pull request by Karan-Palan**
### Implementation

**Rule ID**: multiple_of_default
**Description**: "Setting multipleOf to 1 does not add any further constraint"
**Condition**: Detects multipleOf: 1 (integer) or multipleOf: 1.0 (real) in JSON Schema objects
**Transform**: Removes the redundant multipleOf property
**Dialects**: Supports draft4, draft6, draft7, 2019-09, and 2020-12

- Registered the rule in the Readability mode (not common rules) since it removes redundant default values
- Properly categorized alongside other Default rules like AdditionalPropertiesDefault
- Comprehensive Test Coverage: Added 16 test cases across all supported JSON Schema drafts:

### Test cases: 
Positive cases: multipleOf: 1 and multipleOf: 1.0 removal
Negative cases: No change for non-1 values (multipleOf: 2, multipleOf: 0.5)
Edge cases: String values ("1"), zero (0), negative (-1) - all correctly preserved
Context preservation: Tests with various other keywords to ensure proper behavior

### CODE REVIEW

File: src/extension/alterschema/linter/multiple_of_default.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      -> sourcemeta::core::SchemaTransformRule::Result override {
+    return contains_any(
+               vocabularies,
+               {"https://json-schema.org/draft/2020-12/vocab/validation",
+                "https://json-schema.org/draft/2019-09/vocab/validation",
+                "http://json-schema.org/draft-07/schema#",
+                "http://json-schema.org/draft-06/schema#",
+                "http://json-schema.org/draft-04/schema#"}) &&
+           schema.is_object() && schema.defines("multipleOf") &&
+           schema.at("multipleOf").is_number() &&
```
Comment (jviotti) [inline src/extension/alterschema/linter/multiple_of_default.h:25]: I guess you don't need this line if you are already checking the number possibilities afterwards

File: test/alterschema/alterschema_lint_2019_09_test.cc [hunk 2]
```diff
@@ -1661,8 +1661,7 @@ TEST(AlterSchema_lint_2019_09, unnecessary_allof_ref_wrapper_5) {
   sourcemeta::core::JSON document = sourcemeta::core::parse_json(R"JSON({
     "$schema": "https://json-schema.org/draft/2019-09/schema",
     "allOf": [
-      { 
-        "type": "integer",
+      { "type": "integer",
```
Comment (jviotti) [inline test/alterschema/alterschema_lint_2019_09_test.cc:1664]: Unintended change?

Comment (Karan-Palan) [reply test/alterschema/alterschema_lint_2019_09_test.cc:1664]: It was mostly a formatting thing, I'll rebuild and check

**Comment by Karan-Palan** [comment]
@jviotti added the tests:
- Only multipleOf values of exactly 1 (integer) or 1.0 (real) are removed
- Values like 1.2, 2, 0.5, etc. are preserved as they add meaningful constraints
- String values like "1" are also preserved (not removed)

**Review by jviotti — APPROVED** [review]
Awesome!

**Comment by jviotti** [comment]
Just one more thing: you are failing the DCO check: https://github.com/sourcemeta/core/pull/1804/checks?check_run_id=45432550296

Screenshot 2025-07-07 at 10 40 42

The DCO is to confirm you agree to the contribution guidelines.

**Comment by Karan-Palan** [comment]
@jviotti, The commit was signed (I directly made it through github, as I was deleting vscode/settings.json):

image

However the check still failed. I will rewrite the sign off message and force push, same with the other additonalItems PR. 
I'm not sure the DCO should fail in this case though
